### PR TITLE
cAVS updates

### PR DESCRIPTION
**Pull request by lyakh**
Now that SOF base is in "master," https://github.com/nashif/zephyr/pull/10 is replaced with this one. The commits are identical.

### CODE REVIEW

File: dts/xtensa/intel/intel_cavs25.dtsi
Comment (lgirdwood) [inline dts/xtensa/intel/intel_cavs25.dtsi:44]: Do we need to have a comment in here making it obvious this is for TGL-H variant ?

**Review by lgirdwood — APPROVED** [review]
CAVS and SOF parts LGTM.

File: dts/xtensa/intel/intel_cavs25.dtsi
Comment (lyakh) [reply dts/xtensa/intel/intel_cavs25.dtsi:44]: @lgirdwood we didn't have one before, saying, that it was the LP variant. We'll have to create additional boards and change naming anyway. We have more configurations than just one for each cAVS version. Would be good to record this somewhere - should I create an issue on github or do we prefer to track TODO items elsewhere?

File: scripts/west_commands/sign.py
Comment (mbolivar-nordic) [inline scripts/west_commands/sign.py:429]: Use `from zephyr_ext_common import ZEPHYR_BASE` at top of file instead.

File: scripts/west_commands/sign.py
Comment (mbolivar-nordic) [inline scripts/west_commands/sign.py:436]: Hard-coding the module location like this isn't so nice. How about something like storing its location in the cmake cache instead?

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]

File: scripts/west_commands/sign.py
Comment (lyakh) [reply scripts/west_commands/sign.py]: @mbolivar-nordic thanks! Is this version better? It requires https://github.com/thesofproject/sof/pull/3733 to go with it

File: soc/xtensa/intel_adsp/cavs_v15/include/soc/memory.h
Comment (nashif) [inline soc/xtensa/intel_adsp/cavs_v15/include/soc/memory.h:20]: hmm, why do we have mcuboot here in the first place, we never use mcuboot with those platforms, I would just remove the whole thing.

**Review by nashif — CHANGES_REQUESTED** [review]

File: soc/xtensa/intel_adsp/cavs_v15/include/soc/memory.h
Comment (lyakh) [reply soc/xtensa/intel_adsp/cavs_v15/include/soc/memory.h]: @nashif done

File: scripts/west_commands/sign.py
Comment (lyakh) [reply scripts/west_commands/sign.py]: https://github.com/thesofproject/sof/pull/3733 has been merged

**Review by nashif — APPROVED** [review]

**Comment by lyakh** [comment]
@nashif @mbolivar-nordic what's the procedure in Zephyr - should I dismiss both your "changes requested" reviews? In fact I don't understand why GH hasn't removed that review from @nashif after he has approved this PR?

**Comment by nashif** [comment]
> @nashif @mbolivar-nordic what's the procedure in Zephyr - should I dismiss the remaining "changes requested" review or do I have to wait for the author's acknowledgement of the fix?

lets give @mbolivar-nordic a chance to review.

**Comment by nashif** [comment]
@lyakh  this will requires this one #31233

File: scripts/west_commands/sign.py
Comment (nashif) [inline scripts/west_commands/sign.py:432]: would all of this work on windows? We should consider using os.path.join() instead.

**Review by nashif — CHANGES_REQUESTED** [review]

File: scripts/west_commands/sign.py
Comment (lyakh) [reply scripts/west_commands/sign.py:432]: @nashif I thought `str(pathlib.Path(...) / '...')` would do the right thing? It's used everywhere above...

**Review by nashif — APPROVED** [review]

**Review by mbolivar-nordic — APPROVED** [review]
sign.py bits look fine; thanks.

**Pull request merged**
